plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/ui/internal/view/CDKActionProvider.java

```diff
@@ -89,8 +84,11 @@ protected Object adaptToViewItem(IServer server) {
 			if( beh != null ) {
 				adb = (ServiceManagerEnvironment)beh.getSharedData(ServiceManagerEnvironment.SHARED_INFO_KEY);
 			}
-			if( adb != null )
-				return new CDKOpenshiftUtility().findExistingOpenshiftConnection(server, adb);
+			
+			if( adb != null ) {
+				OpenshiftUIModel model = new OpenshiftUIModel();
+				return model.getConnectionWrapperForConnection(new CDKOpenshiftUtility().findExistingOpenshiftConnection(server, adb));
```

Comment:
@jcantrill, @adietish, @fbricon, I agree with Slava. By now i can just add `model.dispose();`, but such code doesn't look good, as i think, and if dispose change, it will lead to even more confusion. Do we need to create one more JIRA for refactoring OpenshiftUIModel to singleton or may i do it in this PR ?